src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1390,60 +1416,20 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // Arguments:
     // - noteNumber - The MIDI note number to be played (0 - 127).
     // - velocity - The force with which the note should be played (0 - 127).
-    // - duration - How long the note should be sustained (in microseconds).
-    void ControlCore::_terminalPlayMidiNote(const int noteNumber, const int velocity, const std::chrono::microseconds duration)
+    // - duration - How long the note should be sustained.
+    void ControlCore::_terminalPlayMidiNote(const int noteNumber, const int velocity, const MidiDuration duration)
     {
-        // We create the audio instance on demand, and lock it for the duration
-        // of the note output so it can't be destroyed while in use.
-        auto& midiAudio = _getMidiAudio();
-        midiAudio.Lock();
-
         // We then unlock the terminal, so the UI doesn't hang while we're busy.
         auto& terminalLock = _terminal->GetReadWriteLock();
         terminalLock.unlock();
 
         // This call will block for the duration, unless shutdown early.
-        midiAudio.PlayNote(noteNumber, velocity, duration);
+        _midiAudio.PlayNote(reinterpret_cast<HWND>(_owningHwnd), noteNumber, velocity, duration);
 
         // Once complete, we reacquire the terminal lock and unlock the audio.
         // If the terminal has shutdown in the meantime, the Unlock call
         // will throw an exception, forcing the thread to exit ASAP.
         terminalLock.lock();
-        midiAudio.Unlock();

```

Comment:
Again this comment doesn't make sense with the Unlock code removed. Maybe just drop everything after "reacquire the terminal lock".